### ISSUE TITLE
Pass vectors as arguments to function within a function using mapply? (R)

### ISSUE THREAD

**Issue by kubu4** [RobertsLab/resources#221]
I am trying to pass two separate vectors into a function within a function. I'm trying to accomplish this via the ```mapply``` function.

Here's my code (explained in more detail below):

```
# Create function to calculate total alkalinity for all samples.
# Use dplyr library to filter data for use in seacarb library:
# temperature data (T) and convert to vector (.$T)
# potential data (T) and convert to vector (.$E)
# volume data (V) and convert to vector (.$V)
TA_calcs <- function(df, salinities, sample_weights) {
  T_data <- df %>% filter(E <= pH3.0 & E >= pH3.5) %>% select(T) %>% .$T
  E_data <- df %>% filter(E <= pH3.0 & E >= pH3.5) %>% select(E) %>% .$E
  volume_data <- df %>% filter(E <= pH3.0 & E >= pH3.5) %>% select(V) %>% .$V
  mol_to_umol*(at(S=salinities, T=T_data, C=A10_concentration, d=A10_density, weight=sample_weights, E=E_data, volume=volume_data))
  return(df)
}


mapply(TA_calcs, sample_names_list, salinities, sample_weights)
```

Subsequent error message:

```
Error in z$volume * d : non-numeric argument to binary operator
```

Here's a bit of my input data frame:

```
> head(sample_names_list[[1]])
       V t     E    T  dV/dT      S  weight
1 1.5050 0 190.0 21.7     NA 33.481 50.0378
2 1.5065 1 189.5 21.6 0.0005 33.481 50.0378
3 1.5095 2 189.3 21.6 0.0015 33.481 50.0378
4 1.5135 3 189.8 21.6 0.0025 33.481 50.0378
5 1.5195 4 190.3 21.6 0.0035 33.481 50.0378
6 1.5280 5 191.2 21.6 0.0050 33.481 50.0378
```

And, here are these two vectors:

```
> salinities
[1] 33.481 35.000 33.680 33.723 33.990 34.090
> sample_weights
[1] 50.0378 50.0383 50.0409 50.0183 50.0696 50.0576
```

This is the function embedded in my custom function; it's part of the ```seacarb``` library and is used to calculate total alkalinity for a given sample:

```at(S=salinities, T=T_data, C=A10_concentration, d=A10_density, weight=sample_weights, E=E_data, volume=volume_data)```

Finally, the main reason I'm trying to use ```mapply``` is that I'm trying to get this to work on a list of data frames.

In plain English, this is what I'd like to have happen:

For each data frame in my list:

- Subset data for temperature (T), voltage (E), volume (V) and save each to their own vector.

- Pass these vectors to the corresponding argument in the ```seacarb at()``` function.

- Pass each <em>value</em> in the two vectors, ```salinities``` and ```sample_weights```, to the corresponding argument in the ```seacarb at()``` function, one at a time (i.e. "loop" through the vector elements).

Well, now that I've typed all of this out, I'm realizing I can just get it done with a for loop. However, if I could use some version of the various ```apply()``` functions, things would be much cleaner and easier to follow.

Thanks for any suggestions/fixes!

**Comment by kubu4** [comment]
I just noticed something else:

What am I doing with the ```return(df)``` function at the end of my ```TA_calcs``` function? I followed some example I found, but now I'm not sure I need/want that!

**Comment by kubu4** [comment]
Hmmm, interesting development, I get the same error when I run this as a for loop:

```
TA_calcs <- list()
for (item in 1:length(sample_names_list)){
  T_data <- sample_names_list[[item]] %>% filter(E <= pH3.0 & E >= pH3.5) %>% select(T) %>% .$T
  E_data <- sample_names_list[[item]] %>% filter(E <= pH3.0 & E >= pH3.5) %>% select(E) %>% .$E
  volume_data <- sample_names_list[[item]] %>% filter(E <= pH3.0 & E >= pH3.5) %>% select(V) %>% .$V
  salinity <- sample_names_list[[item]][1,"S"]
  weight <- sample_names_list[[item]][1, "weight"]
  TA_calcs <- mol_to_umol*(at(S=salinity, T=T_data, C=A10_concentration, d=A10_density, weight=weight, E=E_data, volume=volume_data))
}
```

```Error in z$volume * d : non-numeric argument to binary operator```

This makes me think that the error message is being thrown by the ```seacarb at()``` function. That would also explain where the ```z$volume``` notation comes into play, as that's not part of any of my data...

I guess it's suggesting that my volume data is non-numeric. Guess I'll investigate further.
